src/components/Comments/DeleteComment.jsx

```diff
@@ -0,0 +1,31 @@
+import { useState } from "react";
+import { deleteComment } from "../utils/api";
+
+const DeleteComment = props => {
+  const { setComments, comment_id } = props;
+  const [isLoading, setIsLoading] = useState(false)
+  const [isError, setIsError] = useState(false)
+
+  const handleClick = () => {
+    setIsLoading(true)
+    deleteComment(comment_id)
+    .then(() => {
+      setIsLoading(false)
+      setComments(comments => comments.filter(comment => comment.comment_id !== comment_id))
+    })
+    .catch(() => {
+      setIsLoading(false)
+      setIsError(true)
+    })
+
+  }
+
+  return (
+    <>
+      {isError && <p className="err-msg">Error Occured: Please try again later</p>}
+      <button disabled={isLoading} onClick={handleClick} className="delete-comment">Delete comment</button>
```

Comment:
Nice implementation of an error message - and stopping the user from being able to mash the button while the API request is going through